From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: braindead <totsuka.sama@gmail.com>
Date: Sat, 5 Nov 2022 17:47:26 -0400
Subject: [PATCH] fix MC-252817 (green map markers do not disappear).

this bug is caused by the fact that the itemframe's item is set to empty before the green marker is requested to be removed. this is fixed by getting the mapid from this method's parameter, rather than the air block now stored by the item frame.

diff --git a/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java b/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
index b42c060a5d8d68b5773a8a5e38c59707a277d9bb..dcf245387f59ce730cb2cfb5fc0e837a20d3dfe5 100644
--- a/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
+++ b/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
@@ -296,7 +296,9 @@ public class ItemFrame extends HangingEntity {
     // Paper end
 
     private void removeFramedMap(ItemStack itemstack) {
-        this.getFramedMapId().ifPresent((i) -> {
+        // Paper start - fix MC-252817 (green map markers do not disappear)
+        this.getFramedMapIdFromItem(itemstack).ifPresent((i) -> {
+            // Paper end
             MapItemSavedData worldmap = MapItem.getSavedData(i, this.level());
 
             if (worldmap != null) {
@@ -314,7 +316,12 @@ public class ItemFrame extends HangingEntity {
 
     public OptionalInt getFramedMapId() {
         ItemStack itemstack = this.getItem();
+        // Paper start - fix MC-252817 (green map markers do not disappear)
+        return this.getFramedMapIdFromItem(itemstack);
+    }
 
+    public OptionalInt getFramedMapIdFromItem(ItemStack itemstack) {
+        // Paper end
         if (itemstack.is(Items.FILLED_MAP)) {
             Integer integer = MapItem.getMapId(itemstack);
 
